src/lib/bakeryjs/Box.ts

```diff
@@ -305,22 +308,50 @@ abstract class Box implements BoxInterface {
 
 	private async processGenerator(value: DataMessage): Promise<any> {
 		try {
+			let siblingsCount = 0;
+			// Prevent the queue from being pushed after generator has resolved
+			const {guardQueue} = BoxEvents(this);
+			const guardedQ = guardQueue(this.queue);
+
 			const retValue: any = await this.processValue(
 				value.getInput(this.meta.requires),
-				(chunk: MessageData[], priority?: number) =>
-					this.queue.push(
+				(chunk: MessageData[], priority?: number) => {
+					siblingsCount += chunk.length;
+					guardedQ.push(
 						chunk.map((msg) => {
 							const parent: Message = value.create();
 							parent.setOutput(this.meta.provides, msg);
 							return parent;
 						}),
 						priority
-					)
+					);
+				}
 			);
 
-			this.queue.push(new SentinelMessage(retValue, value));
+			guardedQ.revoke();
+			this.queue.push(value.createSentinel(siblingsCount, retValue));
 			return;
 		} catch (error) {
+			if (
+				error instanceof TypeError &&
+				error.message.includes('revoked')
+			) {
+				throw new VError(
+					{
+						name: 'GeneratorMisbehaveException',
+						info: {
+							mode: 'generator',
+							box: {
+								name: this.name,
+								meta: this.meta,
+							},
+							value: value.getInput(this.meta.requires),
+						},
+					},
+					'Generator %s emitted new messages after its promise has been resolved, fix the code.',
```

Comment:
`fix the code` zní docela pasivně agresivně, možná bych to vynechal? 😄

Reply:
Cilem bylo napovedet, co je potreba udelat k odstraneni chyby.  Prijde mi, ze oznameni o tom, ze generator emitoval po navratu, muze leckoho zanechat v pochybach, proc to je vubec spatne, cim se to ovlivnuje, a co ma udelat, aby vec napravil.

Reply:
Vynechal jsem to z `message`, pridal jsem to (v mirnejsi forme) do `info.description`.